app/src/test/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModelTest.kt

```diff
@@ -135,8 +185,77 @@ class LoginSSOViewModelTest {
 
         runTest { loginViewModel.login(serverConfig) }
 
-        loginViewModel.loginState.loginSSOError shouldBeInstanceOf LoginSSOError.DialogError.GenericError::class
-        (loginViewModel.loginState.loginSSOError as LoginSSOError.DialogError.GenericError).coreFailure shouldBe networkFailure
+        loginViewModel.loginState.loginSSOError shouldBeInstanceOf LoginError.DialogError.GenericError::class
+        (loginViewModel.loginState.loginSSOError as LoginError.DialogError.GenericError).coreFailure shouldBe networkFailure
+    }
+
+    @Test
+    fun `given establishSSOSession is called, when SSOLogin Success, then SSOLoginResult is passed`() {
+        coEvery { getSSOLoginSessionUseCase.invoke(any(), any()) } returns SSOLoginSessionResult.Success(authSession)
+        coEvery { addAuthenticatedUserUseCase.invoke(any(), any()) } returns AddAuthenticatedUserUseCase.Result.Success(userId)
+        coEvery { registerClientUseCase.invoke(any(), any(), any()) } returns RegisterClientResult.Success(client)
+
+        runTest { loginViewModel.establishSSOSession("", "") }
+
+        coVerify(exactly = 1) { loginViewModel.navigateToConvScreen() }
+    }
+
+    @Test
+    fun `given establishSSOSession is called, when SSOLoginSessionResult return InvalidCookie, then SSOLoginResult fails`() {
+        coEvery { getSSOLoginSessionUseCase.invoke(any(), any()) } returns SSOLoginSessionResult.Failure.InvalidCookie
+        coEvery { addAuthenticatedUserUseCase.invoke(any(), any()) } returns AddAuthenticatedUserUseCase.Result.Success(userId)
+        coEvery { registerClientUseCase.invoke(any(), any(), any()) } returns RegisterClientResult.Success(client)
+
+        runTest { loginViewModel.establishSSOSession("", "") }
+        loginViewModel.loginState.loginSSOError shouldBeInstanceOf LoginError.DialogError.InvalidSSOCookie::class
+        coVerify(exactly = 0) { loginViewModel.registerClient(any()) }
+        coVerify(exactly = 0) { loginViewModel.navigateToConvScreen() }
```

Comment:
We need to assert as well that 
- `getSSOLoginSessionUseCase` is being invoked once
- `addAuthenticatedUser` is not being invoked